core/approval/service.go

```diff
@@ -47,110 +123,284 @@ func (s *Service) BulkInsert(ctx context.Context, approvals []*domain.Approval)
 	return s.repo.BulkInsert(ctx, approvals)
 }
 
-func (s *Service) AdvanceApproval(ctx context.Context, appeal *domain.Appeal) error {
-	policy := appeal.Policy
-	if policy == nil {
-		p, err := s.policyService.GetOne(ctx, appeal.PolicyID, appeal.PolicyVersion)
-		if err != nil {
-			return err
-		}
-		policy = p
+func (s *Service) AddApprover(ctx context.Context, approvalID, email string) error {
+	return s.repo.AddApprover(ctx, &domain.Approver{
+		ApprovalID: approvalID,
+		Email:      email,
+	})
+}
+
+func (s *Service) DeleteApprover(ctx context.Context, approvalID, email string) error {
+	return s.repo.DeleteApprover(ctx, approvalID, email)
+}
+
+// UpdateApproval updates an approval.
+func (s *Service) UpdateApproval(ctx context.Context, approvalAction domain.ApprovalAction) (*domain.Appeal, error) {
```

Comment:
seeing the dependecies to other services makes me think update approval still should be part of appealService 🤔

Reply:
yea, let met thing of something, or do you have any suggestion to make this better?

Reply:
moving back update approval to appealService to reduce dependencies